gaphor/SysML/blocks/block.py

```diff
@@ -207,3 +208,58 @@ def lazy_format(operation):
             },
             draw=draw_top_separator,
         )
+
+
+class BehaviorAsBlockItem(Classified, ElementPresentation[Block]):
```

Comment:
Should we introduce different elements for things represented as block items? We could also add an attribute to the default element and define a different shape.

Reply:
I suspect you do not mean "model element" here. Am I correct? Do you mean a base item which can be extended from to create items that are block shaped?

Please elaborate bit more :slightly_smiling_face:

Reply:
I thought about this in the beginning actually.

My first concern was that ActivityItem is defined under UML for UML activities. And it makes sense that it is represented as it was and I would not want to couple UML with SysML in a sense that a part of UML has to be aware that "bdd" exists.

I really like how dispatch and registering is used to avoid throwing in if-else statements to customize the function behavior. I saw that we can definitely use similar approach to give activities their customizable representations without coupling them all together in a single place - especially since the mechanism is already set up.

Moreover, we would have to start thinking about "where do we draw a line?" because in that sense we may as well make a single Item() class and just add more and more if-else statements to represent things. This is bit over the top example, but I think it illustrates the point. 

Activity as a block is SysML construct. Actually, it is SysML diagram construct - diagram construct to be more specific. The items are kinda like what elements are to the model, but for the diagram. So, model consists of elements; diagram consists of items; model may contain diagrams, which are model elements. And depending on the diagram type, the set of items is technically defined. Registering them would make it more explicit and more easily definable on what diagrams can have what items. This information can later be used for validation and/or to guide proper SysML usage.

We want to be approachable from both UML modellers and SysML modellers. I fear that when UML modeller would see "Show as block" option and be confused because that's not UML specific thing. So your second suggestion would be more appropriate. As a SysML modeller, I would like to be able to represent my behavior architectures similarly how I represent other architectures on the bdd; and feel familiarity with other tools. I would expect the default to be "ActivityAsBlock" instead of having to select "Show as a block".

Putting it all together, I understand now your suggestion and I thought about it, but I had the concerns/reasoning that I explained above. In short, I feel as if your suggestion is more as a "hack" to increase productivity. It definitely would meet my needs and enable me to show activities as a block on bdd, so it is not bad suggestion. If we look more into the future; I think it wont hurt to be able to define diagram explicit items; and leave the door to query a list of elements and their possible representations on the different kinds of diagrams. :slightly_smiling_face:

Reply:
You're right. I did not consider Activity(Item) to be a UML construct.

We need to have a conditional somewhere that determines the shape, and the dispatch functionality can do this job.

Thanks for clarifying. Let's do it as you propose.